### ISSUE TITLE
Convert NLU from non-Markdown to YAML

### ISSUE THREAD

**Issue by pheel** [RasaHQ/rasa#7077]
**Rasa version**: 2.0.0
**Python version**: 3.7.7
**Operating system**: OSX
**Issue**:

**TLDR:** My main question is why is non-Markdown to YAML conversion not supported? [This PR](https://github.com/RasaHQ/rasa/pull/6864) leverages existing code to implement it, but the result is lossy: metadata doesn't make its way into the YAML, because `rasa.shared.nlu.training_data.formats.RasaYAMLWriter` doesn't write metadata.

Is anyone working on making RasaYAMLWriter write metadata? If not I can implement it.

### Steps that got me here

1. `rasa data convert nlu -f yaml --data nlu.json --out nlu.yml`
     > Skipped file: 'nlu.json'.
     Didn't convert any files under 'nlu.json' path. Did you specify the correct file/directory?

     `rasa data convert nlu --help`
     > -f {json,md,yaml}, --format {json,md,yaml}
        Output format the training data should be converted into. **Note: currently training data can be converted to 'yaml' format only from 'md' format** (default: yaml)

     OK, not supported.

2. `rasa data convert nlu -f md --data nlu.json --out nlu.md`
     nlu.md is created successfully, losing any metadata (since Md format lacks metadata support)

3. `rasa data convert nlu -f yaml --data nlu.md --out nlu.yml`
     > The output path 'nlu.yml' doesn't exist. Please make sure to specify an existing directory and try again.

     `touch nlu.yml && rasa data convert nlu -f yaml --data nlu.md --out nlu.yml`
     > NotADirectoryError: [Errno 20] Not a directory: 'nlu.yml/nlu_converted.yml'

     OK, `-f yaml` works the opposite way from `-f json/md`, which gives `IsADirectoryError: [Errno 21]` errors when doing e.g. `--out .`

4. `rasa data convert nlu -f yaml --data nlu.md --out .`
     > Converted NLU file: 'nlu.md' >> 'nlu_converted.yml'.
     Converted 1 file(s), saved in '.'.

### Bonus:

5. **Input:**
    ```yaml
	nlu:
	- intent: estimate_emissions
	  # Arbitrary metadata
	  metadata:
	    author: Some example metadata!
	    key: value
	  # Multiline examples, each line is a separate training example.
	  examples: |
	    - how much CO2 will that use?
	    - how much carbon will a one way flight from [new york]{"entity": "city", "role": "from"} to california produce?
     ```

    `rasa data convert nlu -f json --data nlu.yml --out nlu.json`

    **Output:**
	```json
	{
	  "rasa_nlu_data": {
	    "common_examples": [
	      {
	        "text": "how much CO2 will that use?",
	        "intent": "estimate_emissions",
	        "metadata": {
	          "intent": {
	            "author": "Some example metadata!",
	            "key": "value"
	          }
	        }
	      },
	      {
	        "text": "how much carbon will a one way flight from new york to california produce?",
	        "intent": "estimate_emissions",
	        "entities": [
	          {
	            "start": 43,
	            "end": 51,
	            "value": "new york",
	            "entity": "city",
	            "role": "from"
	          }
	        ],
	        "metadata": {
	          "intent": {
	            "author": "Some example metadata!",
	            "key": "value"
	          }
	        }
	      }
	    ],
	    "regex_features": [],
	    "lookup_tables": [],
	    "entity_synonyms": []
	  }
	}
	```
     Strangely, output above has the metadata embedded under an `intent` key.

**Comment by sara-tagger** [comment]
Thanks for submitting this feature request 🚀 @dakshvar22 will get back to you about it soon! ✨
